pkg/compute/kube/client.go

```diff
@@ -284,3 +264,126 @@ func (c *client) GetHostnamesForService(svc service.MeshService, localNamespace
 
 	return hostnames
 }
+
+// ListServiceIdentitiesForService lists ServiceAccounts associated with the given service
+func (c *client) ListServiceIdentitiesForService(name, namespace string) ([]identity.ServiceIdentity, error) {
+	var identities []identity.ServiceIdentity
+
+	k8sSvc := c.kubeController.GetService(name, namespace)
+	if k8sSvc == nil {
+		return nil, fmt.Errorf("Error fetching service %s/%s: %s", name, namespace, errServiceNotFound)
+	}
+
+	svcAccountsSet := mapset.NewSet()
+	pods := c.kubeController.ListPods()
+	for _, pod := range pods {
+		svcRawSelector := k8sSvc.Spec.Selector
+		selector := labels.Set(svcRawSelector).AsSelector()
+		// service has no selectors, we do not need to match against the pod label
+		if len(svcRawSelector) == 0 {
+			continue
+		}
+		if selector.Matches(labels.Set(pod.Labels)) {
+			podSvcAccount := identity.K8sServiceAccount{
+				Name:      pod.Spec.ServiceAccountName,
+				Namespace: pod.Namespace, // ServiceAccount must belong to the same namespace as the pod
+			}
+			svcAccountsSet.Add(podSvcAccount)
+		}
+	}
+
+	for svcAcc := range svcAccountsSet.Iter() {
+		identities = append(identities, svcAcc.(identity.K8sServiceAccount).ToServiceIdentity())
+	}
+	return identities, nil
+}
+
+// GetService returns the service.MeshService corresponding to the Port used by clients
+// to communicate with it.
+func (c *client) GetService(name, namespace string, port uint16) (service.MeshService, error) {
```

Comment:
The naming here aligns with rest of package, IMO something like GetMeshService() helps mentally keep track where I am in the code base. Not blocking just a comment as I am learning the abstractions

Reply:
Fixed!